services/thermal_mgr/thermal_mgr.c

```
@@ -20,41 +23,104 @@ static StackType_t thermalMgrTaskStack[THERMAL_MGR_STACK_SIZE];
-static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH * THERMAL_MGR_QUEUE_ITEM_SIZE];
+static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH *
+                                          THERMAL_MGR_QUEUE_ITEM_SIZE];
 
 static void thermalMgr(void *pvParameters);
 
 void initThermalSystemManager(lm75bd_config_t *config) {
   memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
   memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
-  
-  thermalMgrTaskHandle = xTaskCreateStatic(
-    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
-    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+
+  thermalMgrTaskHandle =
+      xTaskCreateStatic(thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
+                        config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
 
   memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
   memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
 
-  thermalMgrQueueHandle = xQueueCreateStatic(
-    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
-    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
-
+  thermalMgrQueueHandle =
+      xQueueCreateStatic(THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
+                         thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
-  return ERR_CODE_SUCCESS;
+  // We check if the event was succesfully queued and return accordingly
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) == pdPASS) {
+    return ERR_CODE_SUCCESS;
+    // We check if the queue is full and return accordingly
+  } else if (xQueueIsQueueFullFromISR(thermalMgrQueueHandle) == pdFALSE) {
+    return ERR_CODE_QUEUE_FULL;
+    // If none of the previous cases occured, then we do not know the error and
+    // we return an ERR_CODE_UNKNOWN
+  } else {
+    return ERR_CODE_UNKNOWN;
+  }
```

`xQueueSend` only returns fail if the queue is full so you can return that error code if it fails without needing the else if

Also adding that else if causes an issue where if the queue was full when we ran the if clause but emptied out by the time we ran the else if we will get `ERR_CODE_UNKNOWN` even though the failure was actually because the queue was full when we tried to send. I'd reccomend looking at RTOS 101 to understand better what I mean by race condition